### PR TITLE
README.md: introduce known issue section

### PR DESCRIPTION
**Pull request by fuweid**
The users might run into data corrupted issues caused by underlay filesystem. It's out of scope for bboltdb maintainers to fix filesystem issue. But the section to track known issues can help users and contributors to analyse root cause when they run into data corrupted issues.

Closes: https://github.com/etcd-io/bbolt/issues/562

Thanks @fyfyrchik for providing the details!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -934,6 +934,14 @@ The best places to start are the main entry points into Bolt:
...
 them via pull request.
 
+## Known Issue
+
+- Linux kernel releases [ext4: fast commits](https://lwn.net/Articles/842385/)
+  feature in v5.10. It's a new, lighter-weight journaling method to reduce
+  unrelated IO for fsync/fdatasync. However, `fast commits` is new feature and
+  users run into data loss issue after power failure. The data loss cat corrupt
+  boltdb. If you enable fast commits feature on ext4 filesystem, please ensure
+  that the kernel includes related fix patches to fast commit. Details in [issue 562](https://github.com/etcd-io/bbolt/issues/562).
```
Comment (ahrtr) [inline README.md:944]: - Please add an item in the content
- Please also provide a link to the fast commit issue, and also clarify which linux kernel versions contain the fix or a related link for users reference.

File: README.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ New minor versions may add additional features to the API.
     - [LMDB](#lmdb)
   - [Caveats & Limitations](#caveats--limitations)
   - [Reading the Source](#reading-the-source)
+  - [Known Issue](#known-issue)
```
Comment (ahrtr) [inline README.md:72]:
```suggestion
  - [Known Issues](#known-issues)
```

File: README.md (hunk 1 above)
Comment (fuweid) [reply README.md]: Updated. Please take a look. Thanks

File: README.md [hunk 3]
```diff
@@ -934,6 +935,20 @@ The best places to start are the main entry points into Bolt:
...
+  unrelated IO for fsync/fdatasync. However, `fast commits` is new feature and
+  users run into data loss issue after power failure. The data loss cat corrupt
+  boltdb. There are some related patches.
+
+  * [ext4: fast commit may miss tracking unwritten range during ftruncate](https://lore.kernel.org/linux-ext4/20211223032337.5198-3-yinxin.x@bytedance.com/)
+  * [ext4: fast commit may not fallback for ineligible commit](https://lore.kernel.org/lkml/202201091544.W5HHEXAp-lkp@intel.com/T/#ma0768815e4b5f671e9e451d578256ef9a76fe30e)
+
+  These patches were merged into [kernel v5.17](https://lore.kernel.org/lkml/YdyxjTFaLWif6BCM@mit.edu/).
+  If you enable fast commits feature on ext4 filesystem, please ensure that the
+  kernel includes related fix patches to fast commit. Details in [issue 562](https://github.com/etcd-io/bbolt/issues/562).
```
Comment (ahrtr) [inline README.md:951]:
```suggestion
## Known issues

- bbolt might run into data corruption issue on Linux when the feature
  [ext4: fast commit](https://lwn.net/Articles/842385/), which was introduced in
  linux kernel version v5.10, is enabled. The fixes to the issue were included in
  linux kernel version v5.17, please refer to links below,
  * [ext4: fast commit may miss tracking unwritten range during ftruncate](https://lore.kernel.org/linux-ext4/20211223032337.5198-3-yinxin.x@bytedance.com/)
  * [ext4: fast commit may not fallback for ineligible commit](https://lore.kernel.org/lkml/202201091544.W5HHEXAp-lkp@intel.com/T/#ma0768815e4b5f671e9e451d578256ef9a76fe30e)
  * [ext4 updates for 5.17](https://lore.kernel.org/lkml/YdyxjTFaLWif6BCM@mit.edu/) 

  Please also refer to the discussion in https://github.com/etcd-io/bbolt/issues/562.
```

**Review by ahrtr — APPROVED** [review]

**Comment by fuweid** [comment]
Thanks @ahrtr

**Comment by ahrtr** [comment]
Based on the [Linux kernel version history](https://en.wikipedia.org/wiki/Linux_kernel_version_history), both **5.10** and **5.15** are LTS releases, but 5.17 isn't. So I think we should clearly document the exact 5.10.x and 5.15.x patches in which the fix is included.

**Pull request merged**